### DISCUSSION TITLE
How to conduct local sensitivity analysis with Dakota

### DISCUSSION THREAD

**Discussion by k-ingles** [snl-dakota/dakota#142] [Q&A] [answered]
It is reference very tangentially that the gradient information in the `responses` section can be used to conduct a local sensitivity analysis, but clicking through the methods section, I don't see an option that is dedicated to a local inference. That being said, I didn't know exhaust my options, but I spent the better half of an afternoon looking into this. 
So now I am here to look for help.
Thank you

**Comment by jadamstephens** [comment, marked as answer]
Dakota emphasizes global sensitivity analysis methods, and it doesn't have any dedicated methods for local sensitivity. However, you can ask it to compute gradients at arbitrary points in the input space using the [list_parameter_study](https://snl-dakota.github.io/docs/6.20.0/users/usingdakota/reference/method-list_parameter_study.html) method and setting the gradient specification in your responses to something besides `no_gradients`.

E.g.

```
method
   list_parameter_study
     list_of_points 0.0 0.0

variables
  normal_uncertain 2
    means 0.0 0.0
    std_deviations 1.0 1.0

responses
  response_functions 1
  analytic_gradients
  no_hessians

interface
  direct
    analysis_drivers 'text_book'
```

Produces:

```
---------------------
Begin Evaluation    1
---------------------
Parameters for evaluation 1:
                      0.0000000000e+00 nuv_1
                      0.0000000000e+00 nuv_2

Direct interface: invoking text_book

Active response data for evaluation 1:
Active set vector = { 3 } Deriv vars vector = { 1 2 }
                      2.0000000000e+00 response_fn_1
 [ -4.0000000000e+00 -4.0000000000e+00 ] response_fn_1 gradient
```

**Reply by k-ingles** [reply]
I see, thank you! A follow up question: does the `variables` section actually do anything here? It seems like this input file should only pass the `list_of_points` to the `analysis_drivers`.

**Reply by jadamstephens** [reply]
Most of the time it's true that the variables specification isn't doing anything in a list parameter study. There are a few situations where it does, though. 

- If you specified bounds for your variables and chose a point in your list parameter study that is on an edge or in a corner of the input space, I believe Dakota would perform the finite differencing in a way that avoided violating the bounds.
- A limitation of Dakota's input parser is that arguments to keywords must be all of the same type. String-valued variables must be referred to into the list of points by index. The user provides the actual strings as part of the variables spec.

But, mostly, simple parameter studies are the exception. Virtually all other Dakota methods truly do need a variables specification to work. It doesn't make a lot of sense to carve out a special way of working just for parameter studies.
